File: Python/formulaire.py
class Personne:
    def __init__(self, nom, prenom, sexe, age):
        self.nom = nom
        self.prenom = prenom
        self.sexe = sexe
        self.age = age


class Billet:
    def __init__(self, prix):
        self.prix = prix
        self.prix_reduit = 0.0


def main():
    personne = saisir_informations_personne()  # Saisie des informations de la personne
    billet = saisir_informations_billet()  # Saisie des informations du billet

    calculer_reduction(personne, billet)  # Calcul de la réduction applicable
    reduction = calculer_pourcentage_reduction(billet.prix_reduit)  # Calcul du pourcentage de réduction

    afficher_informations(personne, billet, reduction)  # Affichage des informations


def saisir_informations_personne():
    nom = input("Entrez votre nom: ")
    prenom = input("Entrez votre prénom: ")
    sexe = input("Entrez votre sexe (M ou F): ")
    age = int(input("Entrez votre âge: "))

    return Personne(nom, prenom, sexe, age)  # Création et retour de l'objet Personne


def saisir_informations_billet():
    prix = float(input("Entrez le prix du billet: "))

    return Billet(prix)  # Création et retour de l'objet Billet


def calculer_reduction(personne, billet):
    if personne.age < 26:  # Si l'âge de la personne est inférieur à 26 ans
        if personne.sexe.upper() == "M":
            billet.prix_reduit = 0.7  # Réduction de 30% pour les hommes de moins de 26 ans
        elif personne.sexe.upper() == "F":
            billet.prix_reduit = 0.6  # Réduction de 40% pour les femmes de moins de 26 ans
    else:
        if personne.sexe.upper() == "M":
            billet.prix_reduit = 0.85  # Réduction de 15% pour les hommes de 26 ans et plus
        elif personne.sexe.upper() == "F":
            billet.prix_reduit = 0.8  # Réduction de 20% pour les femmes de 26 ans et plus


def calculer_pourcentage_reduction(prix_reduit):
    return 100 - prix_reduit * 100  # Calcul du pourcentage de réduction


def afficher_informations(personne, billet, reduction):
    genre = "Monsieur" if personne.sexe.upper() == "M" else "Madame"  # Détermination du genre en fonction du sexe

    print(f"Bonjour {genre}, voici vos informations détaillées:")
    print(f"Sexe: {personne.sexe}")
    print(f"Âge: {personne.age} ans")
    print(f"Nom: {personne.nom}")
    print(f"Prénom: {personne.prenom}")
    print(f"Réduction: {reduction:.2f}%")
    print(f"Prix réduit du billet: {billet.prix_reduit:.2f}")


if __name__ == "__main__":
    main()  # Appel de la fonction principale

File: C/formulaire.c
#include <stdio.h>
#include <locale.h>

struct Personne {
    char nom[30];
    char prenom[30];
    char sexe;
    int age;
};

struct Billet {
    float prix;
    double prixReduit;
};

int main() {
    setlocale(LC_ALL, "fr_FR.UTF-8"); // Définit le paramètre régional en français avec le codage UTF-8

    struct Personne personne;
    struct Billet billet = { 0 }; // Initialise billet avec des valeurs nulles

    // Saisie des informations de la personne
    printf("Entrez votre sexe (\"M\" ou \"F\"): ");
    scanf_s(" %c", &personne.sexe, (unsigned int)sizeof(personne.sexe));

    printf("Entrez votre âge: ");
    scanf_s("%d", &personne.age);

    printf("Entrez votre nom: ");
    scanf_s("%s", personne.nom, (unsigned int)sizeof(personne.nom));

    printf("Entrez votre prénom: ");
    scanf_s("%s", personne.prenom, (unsigned int)sizeof(personne.prenom));

    // Calcul de la réduction en fonction de l'âge et du sexe
    if (personne.age < 26) {
        if (personne.sexe == 'M' || personne.sexe == 'm') {
            billet.prixReduit = 0.7; // 30% de réduction pour les hommes de moins de 26 ans
        }
        else if (personne.sexe == 'F' || personne.sexe == 'f') {
            billet.prixReduit = 0.6; // 40% de réduction pour les femmes de moins de 26 ans
        }
    }
    else {
        if (personne.sexe == 'M' || personne.sexe == 'm') {
            billet.prixReduit = 0.85; // 15% de réduction pour les hommes de plus de 26 ans
        }
        else if (personne.sexe == 'F' || personne.sexe == 'f') {
            billet.prixReduit = 0.8; // 20% de réduction pour les femmes de plus de 26 ans
        }
    }

    double reduction = 100 - billet.prixReduit * 100; // Calcul du pourcentage de réduction

    // Saisie du prix du billet
    printf("Entrez le prix du billet: ");
    scanf_s("%f", &billet.prix);

    // Calcul du prix réduit
    billet.prixReduit = billet.prix * billet.prixReduit;

    // Affichage des informations de la personne et du billet
    printf("Bonjour %s, voici vos informations détaillées:\n", (personne.sexe == 'M' || personne.sexe == 'm') ? "Monsieur" : "Madame");
    printf("Sexe: %c\n", personne.sexe);
    printf("Âge: %d ans\n", personne.age);
    printf("Nom: %s\n", personne.nom);
    printf("Prénom: %s\n", personne.prenom);
    printf("Réduction: %.2f%%\n", reduction);
    printf("Prix réduit du billet: %.2f\n", billet.prixReduit);

    return 0;
}
